vscode-ext/src/tests/e2e.ts

```diff
@@ -0,0 +1,130 @@
+import { existsSync } from "fs";
+import { assert, expect } from "chai";
+import { join } from "path";
+import { VSBrowser, StatusBar, TextEditor, EditorView, ActivityBar, DebugView, InputBox, DebugToolbar } from "vscode-extension-tester";
+import get from "axios";
+import { Breakpoint } from "vscode";
+
+
+// This suite tests basic flow of mirroring traffic from remote pod
+// - Enable mirrord -> Disable mirrord
+// - Create mirrord config by pressing the gear icon
+// - Set a breakpoint in the python file
+// - Start debugging the python file
+// - Select the pod from the QuickPick
+// - Send traffic to the pod
+// - Tests successfully exit if breakpoint is hit
+const kubeService = process.env.KUBE_SERVICE;
+const podToSelect = process.env.POD_TO_SELECT;
+
+describe("mirrord sample flow test", function () {
+
+    this.timeout(1000000); // --> mocha tests timeout
+
+    let browser: VSBrowser;
+    let statusBar: StatusBar;
+    let debugToolbar: DebugToolbar;
+
+    const testWorkspace = join(__dirname, '../../test-workspace');
+    const fileName = "app_flask.py";
+    const mirrordConfigPath = join(testWorkspace, '.mirrord/mirrord.json');
+    const configurationFile = "Python: Current File";
+    const breakPoint = 9;
+
+    before(async function () {
+        console.log("podToSelect: " + podToSelect);
+        console.log("kubeService: " + kubeService);
+        expect(podToSelect).to.not.be.undefined;
+        expect(kubeService).to.not.be.undefined;
+        browser = VSBrowser.instance;
+        // need to bring the flask app in open editors
+        await browser.openResources(testWorkspace, join(testWorkspace, fileName));
+        await sleep(10000);
```

Comment:
Why all these sleeps? Seems very prone to breakage, doesn't the testing library provide a way to poll for changes?

Reply:
I think the pattern I have already used in the code `this.browser.wait` should eliminate a lot of sleeps